### PR TITLE
Enable account level quota.

### PR DESCRIPTION
**Pull request by ankagrawal**

### CODE REVIEW

File: ambry-api/src/main/java/com/github/ambry/account/Account.java [hunk 1]
```diff
@@ -74,6 +76,19 @@
  * </p>
  */
 public class Account {
+  public static final QuotaResourceType QUOTA_RESOURCE_TYPE_DEFAULT_VALUE = QuotaResourceType.CONTAINER;
```
Comment (justinlin-linkedin) [inline ambry-api/src/main/java/com/github/ambry/account/Account.java:79]: nit: please put this line after line 106, that's where all the default values stay.

File: ambry-api/src/main/java/com/github/ambry/account/Account.java [hunk 2]
```diff
@@ -143,6 +144,11 @@ private Account(JSONObject metadata) throws JSONException {
           }
           updateContainerMap(containers);
         }
+        if (metadata.has(QUOTA_RESOURCE_TYPE_KEY)) {
+          quotaResourceType = QuotaResourceType.valueOf(metadata.getString(QUOTA_RESOURCE_TYPE_KEY));
+        } else {
+          quotaResourceType = QUOTA_RESOURCE_TYPE_DEFAULT_VALUE;
+        }
```
Comment (justinlin-linkedin) [inline ambry-api/src/main/java/com/github/ambry/account/Account.java:151]: you can do 
```
quotaResourceType = metadata.optEnum(QuotaResourceType.class, QUOTA_RESOURCE_TYPE_KEY, QUOTA_RESOURCE_TYPE_DEFAULT_VALUE);
```

**Review by justinlin-linkedin — APPROVED** [review]

File: ambry-api/src/main/java/com/github/ambry/account/Account.java (hunk 1 above)
Comment (ankagrawal) [reply ambry-api/src/main/java/com/github/ambry/account/Account.java]: done.

File: ambry-api/src/main/java/com/github/ambry/account/Account.java (hunk 2 above)
Comment (ankagrawal) [reply ambry-api/src/main/java/com/github/ambry/account/Account.java]: done.

File: ambry-api/src/main/java/com/github/ambry/quota/QuotaResourceType.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.quota;
+
+/**
+ * Type of Ambry resource for which quota can be applied.
+ */
+public enum QuotaResourceType {
+  ACCOUNT, CONTAINER, SERVICE
+}
```
Comment (Arun-LinkedIn) [inline ambry-api/src/main/java/com/github/ambry/quota/QuotaResourceType.java:21]: Might need a newline ending?

**Review by Arun-LinkedIn — APPROVED** [review]

**Pull request merged**